indico/modules/events/registration/util.py

```diff
@@ -492,3 +499,25 @@ def generate_ticket(registration):
 
 def get_ticket_attachments(registration):
     return [('Ticket.pdf', generate_ticket(registration).getvalue())]
+
+
+def import_registrations_from_csv(regform, f, skip_moderation=True, notify_users=False):
+    """Import event registrants from a CSV file into a form."""
+    reader = csv.reader(f)
+    registrations = []
+    for row in reader:
+        try:
+            first_name, last_name, affiliation, position, phone, email = [to_unicode(value).strip() for value in row]
+        except ValueError:
+            raise CSVImportException('Malformed CSV data - please check that the number of columns is correct')
+
+        with db.session.no_autoflush:
+            registrations.append(create_registration(regform, {
+                'email': email,
+                'first_name': first_name.title(),
```

Comment:
http://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/ ;)

Some of the things there are probably not super relevant, but I have the feeling titlecasing the name will screw e.g. many dutch names ("Foo van Bar" etc.) that have lowercase words. Maybe better to only titlecase if the original value is lowercase-only?

Reply:
The problem is actually UPPERCASE NAMES.

Reply:
lowercase-only or uppercase-only then ;)

Reply:
Well, the question is whether we'd rather have people input all kinds of silly stuff (CAPITALS!) or miss a few ones here and there. I can't see any better solution, unfortunately.